sentry-android-core/src/main/java/io/sentry/android/core/AndroidOptionsInitializer.java

```diff
@@ -134,8 +134,11 @@ static void init(
     initializeCacheDirs(context, options);
     options.setEnvelopeDiskCache(new AndroidEnvelopeCache(options));
 
-    final ActivityFramesTracker activityFramesTracker =
-        new ActivityFramesTracker(loadClass, options.getLogger());
+    final IActivityFramesTracker activityFramesTracker =
```

Comment:
I think that's not gonna work if the user manually sets this option through
```kotlin
SentryAndroid.init { options ->
  options.isEnableActivityFramesTracking = false
}
```
If you look at `SentryAndroid.init` - first, we call `AndroidOptionsInitializer.init` and only after that we call `configuration.configure(options)`, so the user's options are set after this.

I think the right way would be that check for this option in `ActivityLifecycleIntegration.register` after the user's init has been called; if it's disabled, we turn the private instance into a NoOp

Reply:
Oh you're right, `configuration.configure(options)` is called after `AndroidOptionsInitializer.init(..)` - which feels a bit weird 😅

Reply:
yeah, in theory we could swap them, but maybe there was a reason for that @marandaneto knows probably. Maybe to read the Manifest values first pre-setup with them to later give the users possibility to override them manually.

Reply:
yes, the options should be "read" from the SDK point of view, and the user has the chance to swap everything if needed.